altmetric/explorer/api/client.py

```diff
@@ -23,6 +26,44 @@ def digest(secret, message):
     return signature
 
 
+def decode_value(value):
+    try:
+        obj = eval(str(value))
+        if type(obj) in (tuple, list, set):
+            if len(obj) > 1:
+                return [decode_value(item) for item in obj]
+            else:
+                return decode_value(obj[0])
+        else:
+            return int(obj)
+    except:
+        return str(value)
+
+
+FILTER_REGEXP = re.compile(r'filter\[(?P<field>\w+)\]')
+
+
+def api_encode(query_string):
```

Comment:
Does this method name make sense? From the implementation is seems like this just parses the query string, it doesn't seem to be doing any encoding...

Reply:
Yeah I guess it is a bit weird.  I was trying to concisely indicate that it took a url and turned it into a data structure that could be used to run the same query using the api client but I guess it was a bit _too_ concise.  How does the new name I pushed look?

Reply:
Yep that's much more understandable!